Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.targets

```diff
@@ -127,18 +133,34 @@
 
     <Message Text="[Reqnroll] Generated code-behind file: %(ReqnrollGeneratedFiles.Identity)" Importance="normal" />
     <Message Text="[Reqnroll] Generated messages file: %(ReqnrollGeneratedFiles.MessagesFile)" Condition="'%(ReqnrollGeneratedFiles.MessagesFile)' != ''" Importance="normal" />
+  </Target>
+
+  <Target Name="IncludeProcessedReqnrollItemsToBuildInputs" DependsOnTargets="CoreProcessReqnrollFeatureFilesInProject">
+    <!-- We need to process the item inclusions in a separate target, because if we process it in the
+         target that has inputs/outputs defined the order of the items will be dependent on what has changed 
+         (changed items will be processed last).
+         This is a problem, because the C# compiler only detects up-to-date state when the order of 
+         the Compile items is the same as in the previous build.
+    -->
```

Comment:
Wasn't aware of this nuance. Presumably some downstream target does an up-to-date check that we'd like to satisfy?

I'm surprised though, even when a target is skipped due to an up-to-date check, the *metadata* changes are supposed to still be applied as though the target was executed.